torch/nested/_internal/ops.py

```diff
@@ -58,17 +58,22 @@ def check_schema(schema_str: str, func, *args, **kwargs) -> None:
     num_optional_args = sum([x.endswith("?") for x in named_arg_types])
     min_args = len(named_arg_types) - num_optional_args
 
-    if not (len(args) >= min_args and len(args) <= len(named_arg_types)):
-        raise ValueError(
-            f"NestedTensor {func.__name__}({schema_str}): expected at least {min_args} "
-            f"arguments and at most {len(named_arg_types)} arguments, but got: "
-            f"{len(args)} arguments"
-        )
+    # special case: ellipses allows for any number of unchecked args at the end
+    if named_arg_types[-1] == "...":
+        named_arg_types = named_arg_types[:-1]
+    else:
+        if not (len(args) >= min_args and len(args) <= len(named_arg_types)):
+            raise ValueError(
+                f"NestedTensor {func.__name__}({schema_str}): expected at least {min_args} "
+                f"arguments and at most {len(named_arg_types)} arguments, but got: "
+                f"{len(args)} arguments"
+            )
 
     arg_type_check_fns = {
         "t": lambda x: isinstance(x, torch.Tensor) and not isinstance(x, NestedTensor),
         "jt": lambda x: isinstance(x, NestedTensor)
-        and x._lengths is None,  # ops with "jt" require contiguous JT only
+        and x._lengths is None
+        and x._ragged_idx == 1,  # ops with "jt" require contiguous JT only
```

Comment:
With this constraint, how do we support the SDPA use case? SDPA requires ragged_idx = 2 and contiguous. But after we transpose the NT to ragged_idx = 2, it can't be made contiguous anymore?